### PR TITLE
feat: only require reindexing when the index was on going to off

### PR DESCRIPTION
**Pull request by PastaPastaPasta**
## What was done?
It does not seem reasonable that we need to reindex when turning indexes off. this logic was introduced in https://github.com/dashpay/dash/commit/e0781095f0d797c978fa25d2e0afb4b6dbffddfe

## How Has This Been Tested?
Hasn't

## Breaking Changes
None, basically

## Checklist:
  _Go over all the following points, and put an `x` in all the boxes that apply._
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone _(for repository code-owners and collaborators only)_

### CODE REVIEW

**Comment by DashCoreAutoGuix** [comment]
Guix Automation has began to build this PR tagged as v21.1.0-devpr6203.c412d905. A new comment will be made when the image is pushed.

**Comment by DashCoreAutoGuix** [comment]
Guix Automation has completed; a release should be present here: https://github.com/dashpay/dash-dev-branches/releases/tag/v21.1.0-devpr6203.c412d905. The image should be on dockerhub soon.

**Comment by PastaPastaPasta** [comment]
Anything else needed here?

**Comment by UdjinM6** [comment]
> Anything else needed here?

https://github.com/dashpay/dash/pull/6203#discussion_r1713702806:
> Will have to update tests too.

**Comment by PastaPastaPasta** [comment]
> > Anything else needed here?
> 
> [#6203 (comment)](https://github.com/dashpay/dash/pull/6203#discussion_r1713702806):
> 
> > Will have to update tests too.

Whoops! I swear I already did that locally, just forgot to push it I guess

File: test/functional/feature_addressindex.py [hunk 1]
```diff
@@ -42,12 +42,12 @@ def setup_network(self):
     def run_test(self):
         self.log.info("Test that settings can't be changed without -reindex...")
         self.stop_node(1)
-        self.nodes[1].assert_start_raises_init_error(["-addressindex=0"], "You need to rebuild the database using -reindex to change -addressindex", match=ErrorMatch.PARTIAL_REGEX)
+        self.nodes[1].assert_start_raises_init_error(["-addressindex=0"], "You need to rebuild the database using -reindex to enable -addressindex", match=ErrorMatch.PARTIAL_REGEX)
```
Comment (UdjinM6) [inline test/functional/feature_addressindex.py:45]: `-addressindex=0` == _disable_
you don't need to reindex to _disable_ it...

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
see below + same for 2 more tests `feature_spentindex.py` and `feature_timestampindex.py`

File: test/functional/feature_addressindex.py [hunk 2]
```diff
@@ -42,12 +42,11 @@ def setup_network(self):
     def run_test(self):
         self.log.info("Test that settings can't be changed without -reindex...")
         self.stop_node(1)
-        self.nodes[1].assert_start_raises_init_error(["-addressindex=0"], "You need to rebuild the database using -reindex to change -addressindex", match=ErrorMatch.PARTIAL_REGEX)
-        self.start_node(1, ["-addressindex=0", "-reindex"])
+        self.start_node(1, ["-addressindex=0"])
```
Comment (knst) [inline test/functional/feature_addressindex.py:45]: -self.stop_node(1)
-self.start_node(1, ["-addressindex=0"])
+self.restart_node(1, "[-addressindex=0"])

**Review by knst — COMMENTED** [review]
let's wait CI, also consider replacement `stop+start` to `restart_nodes` which has been refactored in all functional tests long time ago

**Comment by UdjinM6** [comment]
pls consider 7399b7ade5

**Comment by PastaPastaPasta** [comment]
Finally pulled in your commit 🙈 I forget about too many PRs

**Review by UdjinM6 — APPROVED** [review]
utACK 492204dc4e9ad856f5ff8a25f5c6cbb44b8a7fe5

**Comment by PastaPastaPasta** [comment]
no diff; squashed these 4 commits into 1

**Review by knst — APPROVED** [review]
utACK c96f9e028571e3d592c5043f3ff1a6f672cba7f3

**Pull request merged**